flair/rpc/rpc.go

```diff
@@ -245,12 +246,27 @@ func (s *server) BatchUpdateBlobs(ctx context.Context, req *pb.BatchUpdateBlobsR
 				}
 				mutex.Lock()
 				defer mutex.Unlock()
-				resp.Responses = append(resp.Responses, r.Responses...)
+				for _, resp := range r.Responses {
+					m[resp.Digest.Hash] = append(m[resp.Digest.Hash], resp)
+				}
 				return nil
 			})
 		})
 	}
-	return resp, g.Wait()
+	err := g.Wait()
+	chooseResponse := func(v []*pb.BatchUpdateBlobsResponse_Response) *pb.BatchUpdateBlobsResponse_Response {
+		for _, r := range v {
+			if r.Status.Code == int32(codes.OK) {
+				return r
+			}
+		}
+		// Not ideal but just return the first one so it has a sensible code.
+		return v[0]
```

Comment:
I think we'd want to have some observability on when we are hitting this case. Thoughts on a log/counter here?

Reply:
Fair - added a log message. There aren't any metrics in Flair at all rn so I've not gone into that but we might want to think about it later (I suspect that we don't have anything that would actually collect them atm anyway).